### PR TITLE
docs: updated documentation

### PR DESCRIPTION
**Pull request by dharmez**
Updated README.md and DATAFLOW.md

### CODE REVIEW

File: assets/deployment-architecture.png
Comment (eeaton) [inline assets/deployment-architecture.png:1]: I have a number of small comments with the diagram, perhaps it would be easier to review and edit in slides directly instead of Github?

 - unusual abbreviation "bank stmts", write it out as "bank statements"
 -  if we're using bullets to separate items, don't need "," or "&" between items
 - replace "employee (IT) OR Cron Job" with "Uploader persona"
 - replace "empoyee biz ops" with "Reader persona"
 - "Excel, Outlook Processor" isn't called this in the job names, which job does it refer to? 
 - bubble "11" should be between Web UI and the Reader, not between Web UI and Search App
 - text for number 5 references "AlloyDB, BigQuery and Firestore" but the diagram avoids BigQuery, is this deliberate?
 - numbering is off, the diagram has 12 but text has 10. I think the gap is 9,10,12 on diagram don't exist in text.

File: DATAFLOW.md
Comment (eeaton) [inline DATAFLOW.md:9]: I don't think we ever explained the criteria for how documents are rejected. Is it unsupported file types?

File: DATAFLOW.md
Comment (eeaton) [inline DATAFLOW.md:14]: Names of jobs here don't match the names of Jobs that we create, so it's hard to understand which is referring to which

File: DATAFLOW.md
Comment (eeaton) [inline DATAFLOW.md:13]: We use "process" and "parser" interchangeably, but don't explain what either means in this context. Needs more explanation of what it does, maybe something like "processes form documents by reading the document and extracting fields that will be later used for RAG" (I'll defer to you if there's a better description)

File: DATAFLOW.md
Comment (eeaton) [inline DATAFLOW.md:17]: It's hard to follow how steps 5, 6, and 7 are different. They're saying mostly the same thing but highlight different data products, but it's not clear to the reader why the multiple sources are required or what outcomes are enabled by the multiple sources

File: DATAFLOW.md
Comment (eeaton) [inline DATAFLOW.md:24]: "GCS data is imported... to Agent Builder's **BigQuery** Data Store" --> is this correct?
I thought the data store could be either GCS-based or BQ-based depending on the input of source files.  The diagram shows both but the text makes it sound like GCS inputs are converted to a BQ-based datastore. 

To make clearer, how about something that generalizes across both sources like "Data and metadata from the processed documents is imported into an Agent Builder Data Store"

File: README.md
Comment (eeaton) [inline README.md:57]: I disagree with moving the custom classifier steps to here. Most of this section is about how to deploy the Terraform resources self-contained in this repo, so it's out of place to require elaborate steps that aren't part of the repo.

(Same argument for the step about the domain)

I suggest we make this subsectionection only for deploying the Terraform. How about restructuring like...
1. Deployment Guide
    1. Deploy the terraform
    2. Configure the custom classifier (this content)
    3. Enable the web application (steps about the domain name ownership, and additional steps about DNS and OAuth config that currently aren't documented)

File: DATAFLOW.md
Comment (eyalbenivri) [reply DATAFLOW.md:7]: I'm not sure if we explained it or not, but if not, we definitely should. Unsupported file types is one, but we also reject files that are too big, or files that we detected as duplicates.

File: DATAFLOW.md
Comment (eyalbenivri) [reply DATAFLOW.md:22]: I'll just say that documents are kept in GCS, but data extracted from the document is kept in BQ and AlloyDB, plus metadata in BQ, plus ingested in Agent builder, which is the secret sauce (vectorized version of the document).

So combination of all.

File: README.md
Comment (eeaton) [inline README.md:108]: In my previous commit I changed this section back the to the shell code format, and it looks like you reverted it back again to the table. Why the table?

My view is that the lines of shell are easier to copy paste with a few clicks to modify the values. The table format is harder to use because they have to copy paste each line individual or reformat it somewhere else before they can put it in a shell.

File: README.md
Comment (eeaton) [inline README.md:158]: same as other comments about shell vs table for inputting shell variables

File: README.md
Comment (eeaton) [inline README.md:282]: Nit: it's odd to give full sentence directions in a section header, how about the following...

Instead of 
```
### Search and Explore from EKS Web UI (Skip this if you have not configured the access to the Web UI Search Application)
```

I suggest:
```
### Search and Explore from EKS Web UI
_Note: This section only applies if you completed the "configure access to the Web UI search application" step under the deployment guide. 
```

File: README.md
Comment (eeaton) [inline README.md:118]: Line 118 introduced trailing whitespace that causes issues with Lint.

You can fix this and similar issues by running:
```
export LINTER_CONTAINER_FIX_MODE=true
scripts/lint.sh
```

Or, I configure my IDE with a devcontainer so that it will automatically configure the built-in linters to catch and fix this on every save.

**Comment by eeaton** [comment]
I've made a few further comments, they don't need to be blocking so I'll give an approval now but make sure to resolve those and fix lint issues before merging.

**Review by eeaton — APPROVED** [review]
I've made a few further comments, they don't need to be blocking so I'll give an approval now but make sure to resolve those and fix lint issues before merging.

File: README.md
Comment (dharmez) [reply README.md:108]: The users should clearly understand these setting and so I think we should keep this in a table form so that when they do it one at a time, they think rather than blank copy-paste.

File: DATAFLOW.md
Comment (dharmez) [reply DATAFLOW.md]: We are not rejecting the duplicates at the moment.

File: assets/deployment-architecture.png
Comment (dharmez) [reply assets/deployment-architecture.png:1]: I fixed the numbering in the DATAFLOW.md

File: README.md
Comment (dharmez) [reply README.md:282]: Yes, I will make this change.

File: DATAFLOW.md
Comment (dharmez) [reply DATAFLOW.md:4]: Ok, I will update it.

File: README.md
Comment (dharmez) [reply README.md:118]: Ok, thanks!

**Review by eeaton — APPROVED** [review]

**Review by dharmez — COMMENTED** [review]
LGTM

**Review by eeaton — APPROVED** [review]

**Pull request merged**
